domain-management/src/main/java/org/jboss/as/domain/management/security/PropertiesFileLoader.java

```diff
@@ -225,6 +225,41 @@ public synchronized void persistProperties() throws IOException {
         }
     }
 
+    protected String cleanKey(final String key) {
+        char[] keyChars = key.toCharArray();
+        char[] cleaned = new char[keyChars.length];
+        int keyPos = 0;
+        int cleanedPos = 0;
+
+        while (keyPos < keyChars.length) {
+            if (keyChars[keyPos++] == '\\') {
```

Comment:
Should this fail here if keyPos == keyChars.length?

IOW what if the user edits the file manulally and doesn't escape a final '\'?

foo\=group-a

instead of the correct

foo\\=group-a

Currently L236 will fail with an IndexOutOfBoundException instead of an error message explaining the issue.